File: algorithm3/algorithm3.go
// sorry nothing work code, but please see js file
package main

import (
	"fmt"
	"strconv"
)

func expandedForm(x int) {
	var pow []int
	var decimal []int
	x = strconv.Itoa(x)
	var lengths = x.len
	for i := 0; i <= lengths-1; i++ {
		pow.unshift(i)
	}
	for _, foreach := range x {
		x = strconv.Atoa(x)
		decimal.push(x * 10 * *pow[index])
	}

	var toDecimal = decimal.filter(a >= a && a != 0)
	return strconv.Itoa(toDecimal)
}

func main() {
	var input = 1345679
	fmt.Println(expandedForm(input))
}

File: algorithm3/algorithm3.js
const expandedNumber = num => {
    let pow = []
    let decimal = []
    num = num.toString().split('')
    let len = num.length
    for(let i = 0; i <= len - 1; i++){
        pow.unshift(i)
    }
    num.forEach((x,index) => {
        x = parseInt(x)
        decimal.push(x*10**pow[index])
    })
    let toDecimal = decimal.filter(a => a !== 0)
    return toDecimal.toString().split(',').join('\r\n')
};

var input = 1345679
console.log(expandedNumber(input))
